src/python/pants/core/util_rules/archive.py

```diff
@@ -100,6 +100,9 @@ async def create_archive(request: CreateArchive) -> Digest:
             output_dir_digest = await Get(Digest, CreateDigest([Directory(output_dir)]))
             input_digests.append(output_dir_digest)
 
+        # @TODO: Excludelist from immutable input symlinking everything, as `tar` is symlink-aware
+        # See https://github.com/pantsbuild/pants/issues/17712
```

Comment:
s/symlink/permissions/g

Reply:
I'm just gonna remove this block, it makes less sense now that we're using hardlinks.